en/computer_vision/obstacle_avoidance.md

```diff
@@ -0,0 +1,156 @@
+- A waypoint is "reached" when the vehicle is within the acceptance radius, regardless of its heading.
+  - In normal missions the vehicle must reach a waypoint with a certain heading (i.e. in a "close to" straight line from the previous waypoint). 
+  - When obstacle avoidance is active, this constraint cannot be fulfilled because the obstacle avoidance algorithm has full control of the vehicle heading, and the vehicle always moves in the current field of view. 
+- navigator updates the triplets when the vehicle has reached the acceptance radius of each waypoint. 
+  If a waypoint is inside an obstacle it can happen that it’s never reach and the mission will be stuck. 
+  If the vehicle projection on the line previous-current waypoint has passed the current waypoint, the acceptance radius is enlarged such that the current waypoint is set as reached
+- If the vehicle within the x-y acceptance radius, the altitude acceptance is modified such that the mission progresses (even if it is not in the altitude acceptance radius).
+  
+
+###  Mission Mode Avoidance Interface {#mission_avoidance_interface}
+
+Mission mode is enabled on PX4 by setting `MPC_OBS_AVOID` to `1`.
+
+PX4 communicates with the computer vision system on the companion computer using an implementation of the [Path Planning Protocol](TBD) (Trajectory Interface). <!-- check we call it this! -->
+
+PX4 sends the desired trajectory to the companion computer in `TRAJECTORY_REPRESENTATION_WAYPOINTS` messages at 5Hz.
+The "waypoint" array fields are set as shown:
+- _index 0 :_
+  - position: x-y-z NED vehicle local position
+  - velocity: x-y-z NED velocity setpoint
+  - acceleration: vehicle acceleration
+  - yaw: vehicle yaw
+  - yaw_speed: NaN
+- _index 1:_
+  - position: x-y-z NED local coordinates of the current mission waypoint
+  - velocity: NaN
+  - Acceleration: NaN
+  - yaw: yaw setpoint
+  - yaw_speed: yaw speed setpoint
+- _Index2:_
+  - position: x-y-z NED local coordinates of the next mission waypoint
+  - velocity: NaN
+  - acceleration: NaN
+  - yaw: yaw setpoint
+  - yaw_speed: yaw speed setpoint
+- The remaining indices/fields are filled with NaN. 
+
+PX4 expects to receive target setpoints in a stream of `TRAJECTORY_REPRESENTATION_WAYPOINTS` messages.
+The array waypoints should contains all `NaN` except for index 0:
+- _index 0 :_
+  - Position: position setpoint
+  - Velocity: velocity setpoint
+  - acceleration: `NaN` (acceleration setpoints are not supported by PX4)
+  - Yaw: yaw setpoint
+  - Yaw_speed: yaw speed setpoint
+
+The messages should be sent over the whole mission (not just when navigating around an obstacle).
+The rate at which target setpoints are sent depends on the capabilities of the planning software. 
+Nominally this should exceed [TBD].
+
+The paragraphs below describe the behaviour in greater detail, covering the internal PX4 behaviour and message flow through ROS.
+
+#### Mission Mode Detailed Behaviour
```

Comment:
FYI, this just copies the text you provided. I think it will help people who want to understand the parts of the story.

Reply:
Sure :)